### PR TITLE
IQSS/10117-increase-guestbook-question-length

### LINKED ISSUE #9492: Creating a guestbook with a custom question that has more than 255 characters says "success" but silently fails
If you try to create a guestbook with a custom question that has more than 255 characters, the result us the success page, but no new guestbook is created. (in the log we see an exception that gets thrown.

```
org.postgresql.util.PSQLException: ERROR: value too long for type character varying(255)
Error Code: 0
Call: INSERT INTO CUSTOMQUESTION (DISPLAYORDER, HIDDEN, QUESTIONSTRING, QUESTIONTYPE, REQUIRED, GUESTBOOK_ID) VALUES (?, ?, ?, ?, ?, ?)
```

We should do one of the following: 
a) keep the user on the input page with the error message at the top, 
b) limit # of characters that can be input, or 
c) allow longer questions

### LINKED ISSUE #10117: Guestbook custom questions limited to 255 chars and creating one fails silently if you go over that limit
**What steps does it take to reproduce the issue?**
Per the title, trying to create/edit a guestbook with more than 255 chars in a custom question results in a success message but no change. The underlying limit is due to the column type varchar(255). The error failure appears to be due to a missing return statement when the CommandException is caught, allowing the code to create and send the success message to run.

PR to follow

### PR DESCRIPTION
**Pull request by qqmyers**
**What this PR does / why we need it**: This PR increases the allowed length of a custom question (was 255 chars, now up to limit of the postgres text type) and fixes the code that was hiding any error when a guestbook create/edit failed (due to this or other reason causing a CommandException).

**Which issue(s) this PR closes**:

Closes #10117
Closes #9492

**Special notes for your reviewer**: Accidentally committed to develop, reverted everything, and added a new commit here to restore everything.

**Suggestions on how to test this**: Try creating/editing a guestbook with a custom question over 255 chars in length. Before the PR, you'll see a success message and no new guestbook/guestbook update. After the PR, the change will succeed. If you also want to test the error fix, run `ALTER TABLE customquestion ALTER COLUMN questionstring TYPE  varchar(10);` or similar low number and try to add a question longer than that. (setting TYPE back to text as in the flyway script will restore your db).

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**: fairly minor issue but I added one

**Additional documentation**:

### CODE REVIEW

**Comment by cmbz** [comment]
2023/12/18: Also closes [#9492](https://github.com/IQSS/dataverse/issues/9492)

File: src/main/resources/db/migration/V6.1.0.6__10017-failure-with-long-custom-question.sql
Comment (stevenwinship) [inline src/main/resources/db/migration/V6.1.0.6__10017-failure-with-long-custom-question.sql:1]: This file needs to be renamed V6.5....

**Review by stevenwinship — APPROVED** [review]

**Comment by pdurbin** [comment]
@stevenwinship @qqmyers @ofahimIQSS Jenkins tests are failing. Does anyone know why?

**Comment by stevenwinship** [comment]
> @stevenwinship @qqmyers @ofahimIQSS Jenkins tests are failing. Does anyone know why?

[Pipeline] script
[Pipeline] {
[Pipeline] fileExists
[Pipeline] error
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: Ansible run terminated abnormally, failing build.

GitHub has been notified of this commit’s build result

Finished: FAILURE

**Comment by stevenwinship** [comment]
"System is booting up. Unprivileged users are not permitted to log in yet. Please come back later. For technical details, see pam_nologin(8)."
Connection closed by 3.236.18.81 port 22
lost connection

**Comment by ofahimIQSS** [comment]
No issues found during testing - Merging PR
[Testing of 10118.docx](https://github.com/user-attachments/files/17818382/Testing.of.10118.docx)

**Comment by miramgleiber** [comment]
Hi! Has this been fixed? I'm working on a guestbook through Borealis and still have this issue. Thanks!

**Comment by qqmyers** [comment]
It's fixed as of v6.5. Borealis is running a customized 'v6.4.1-SP' and so probably doesn't have the fix yet.

**Pull request merged**